formats/gltf/writer.go

```diff
@@ -369,6 +369,11 @@ func (w *Writer) AddMaterial(mat PolyformMaterial) *int {
 		emissiveFactor = &factor
 	}
 
+	var alphaCutOff *float64
+	if mat.AlphaMode != nil {
+		alphaCutOff = mat.AlphaCutoff // alphaCutOff should only be set if the alphaMode is set
+	}
```

Comment:
I wanted your thoughts on this implementation. I could see users getting confused by us silently not setting `alphaCutOff` even though they included it. Given it's illegal, it might warrant a panic.

```suggestion
	alphaCutOff := mat.AlphaCutoff
	if mat.AlphaMode == nil && mat.AlphaCutoff != nil {
	        // https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_material_alphacutoff
	        panic("When alphaMode is not defined, alphaCutOff MUST NOT be defined")
	}
```

An alternative approach to this would be preventing the user from ever making an illegal move through the introduction of a new field to the PolyformMaterial that encapsulates the alpha logic.

```golang
type PolyformMaterialAlpha interface {
    Set(mat PbrMetallicRoughness)
}

type PolyformMaterialAlphaBlend struct {
     CutOff *float64
}

func (pmab PolyformMaterialAlphaBlend) Set(mat PbrMetallicRoughness) {
    mat.AlphaMode = MaterialAlphaMode_BLEND
    mat.AlphaCutoff = pmab.CutOff
}
```

Appreciate your close attention to the spec. I totally missed that bit of logic.

Reply:
I see your point, but are you sure this is the best way to solve it?

I think, if we treat it as an invalid input - it warrants an error to be bubbled up and returned from the `WriteText/WriteBinary` top level functions. Inducing a `panic` crash seems a bit harsh reaction for an invalid user input 😃

Also, your suggestion will prevent user from setting wrong parameters, but it also makes the interface quite less ergonomic and implementation - more complex.

I think we should strike a balance here between being careless on one hand and overbearing on the other.  

My first implementation was too careless, let me try adding an error return. Let me know what you think.

Reply:
Made an update, let me know what you think of this version.

Reply:
I think this is a good compromise! Thanks again for this fix.